modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -332,7 +355,31 @@ sda.enkf.multisite <- function(settings,
         #furrr::future_map(~ unlink(.x))
         
         #for next time step split the met if model requires
-        inputs.split <- metSplit(conf.settings, inputs, settings, model, no_split, obs.times, t, nens, restart_flag, my.split_inputs)
+        #-Splitting the input for the models that they don't care about the start and end time of simulations and they run as long as their met file.
+        inputs.split <- conf.settings %>%
+          `class<-`(c("list")) %>%
+          purrr::map2(inputs, function(settings, inputs) {
+            # Loading the model package - this is required bc of the furrr
+            library(paste0("PEcAn.",model), character.only = TRUE)
+            
+            inputs.split <- list()
+            if (!no_split) {
+              for (i in seq_len(nens)) {
+                #---------------- model specific split inputs
+                inputs.split$samples[i] <- do.call(
+                  my.split_inputs,
+                  args = list(
+                    settings = settings,
+                    start.time = (lubridate::ymd_hms(obs.times[t - 1], truncated = 3) + lubridate::second(lubridate::hms("00:00:01"))),
+                    stop.time =   lubridate::ymd_hms(obs.times[t], truncated = 3),
+                    inputs = inputs$samples[[i]])
+                )
+              }
+            } else{
+              inputs.split <- inputs
+            }
+            inputs.split
+          })
```

Comment:
In addition to all the points @infotroph brought up, I'm really not a fan of replacing a nice, modular function, `metSplit`, with a big chunk of code that reduces the overall readability of the code. I think this line should be reverted and if there's something inside `metSplit` that needs fixing, then you should fix it internally within that function rather than ditching the function.

Reply:
re-upping this comment so you don't forget to make this change

Reply:
Added comment above this chunk of metSplit part, and I will replace it with the metSplit function later once I completed the debug with it!

Reply:
Added @export inside the metSplit function so the metSplit will be exported in the namespace. Also replace the big chunk of met split code with metSplit function call.